fetcher.go

```diff
@@ -74,14 +75,17 @@ func (p *pool) doFetch(ctx context.Context, from string, c cid.Cid, attempt int)
 }
 
 // TODO Refactor to use a metrics collector that separates the collection of metrics from the actual fetching
-func (p *pool) fetchResource(ctx context.Context, from string, resource string, mime string, attempt int, cb DataCallback) (err error) {
+// rm will be nil only for context cancellation errors
```

Comment:
this is confusing - it looks like it isn't nil even in the context cancellation case - it's just an empty ResponseMetrics objects?

Reply:
Yeah this is a remanant of a previous iteration. Fixed.